src/gen-ts.ts

```diff
@@ -136,6 +139,21 @@ function analyzerToAst(
   return tsDocs;
 }
 
+/**
+ * Analyzer always returns fully specified URLs with a protocol and an absolute
+ * path (e.g. "file:/foo/bar"). Return just the file path, relative to our
+ * project root.
+ */
+function analyzerUrlToRelativePath(
```

Comment:
For this I think you FSUrlLoader's method to get the file path, which will return an absolute path if and only if the given url is a file url that's contained within its root dir. You can then call path.relative on that path.

Or you want the one-argument form of UrlResolver.relative(), then parse the resulting url to be sure that it only has a pathname. But the FSUrlLoader method is nice because it ensures it's a local file url, and that it's within the base path, which means that you can be confident that you'll only write to files inside the root dir.

Reply:
> For this I think you FSUrlLoader's method to get the file path, which will return an absolute path if and only if the given url is a file url that's contained within its root dir. You can then call path.relative on that path.

I need to also resolve paths that are not in the project root, for example `paper-button` has an HTML import for `../polymer/polymer-element.html`, and I need to produce a corresponding `../polymer/polymer-element.d.ts` triple-slash reference.

> Or you want the one-argument form of UrlResolver.relative(), then parse the resulting url to be sure that it only has a pathname.

I'm using the `analyzer.urlResolver.relative`, and it is returning regular file paths (no protocol prefix or anything). Is that what you expected? I just pushed a new commit, but it doesn't do any checking that it's a pathname. PTAL?

Reply:
> I'm using the `analyzer.urlResolver.relative`, and it is returning regular file paths (no protocol prefix or anything). Is that what you expected? I just pushed a new commit, but it doesn't do any checking that it's a pathname. PTAL?

For most projects that will work fine, but for urls that are actually relative paths, like `https://polygit.org/components/polymer/polymer.html`, `UrlResolver#relative` is a noop.